bin/btmon.py

```diff
@@ -3937,7 +3980,61 @@ def _handle_urlopen_error(self, e, url, payload):
                         '\n  api_key: ' + self.api_key,
                         '\n  system_id: ' + self.system_id,
                         '\n  data:  ' + payload]))
-            
+
+class InfluxDBProcessor(UploadProcessor):
+    def __init__(self, host, port, username, password, database, measurement, map_str, period, timeout):
+        super(InfluxDBProcessor, self).__init__()
+        self.host = host
+        self.port = port
+        self.username = username
+        self.password = password
+        self.database = database
+        self.measurement = measurement
+        self.map_str = map_str
+        self.process_period = int(period)
+        self.timeout = int(timeout)
+        self.map = dict()
+
+        infmsg('InfluxDB: upload period: %d' % self.process_period)
+        infmsg('InfluxDB: host: %s' % self.host)
+        infmsg('InfluxDB: port: %s' % self.port)
+        infmsg('InfluxDB: username: %s' % self.username)
+        infmsg('InfluxDB: map: %s' % self.map_str)
+
+    def setup(self):
+        self.map = pairs2dict(self.map_str)
+
+    def process_calculated(self, packets):
+        sensors = dict()
+        readings = dict()
+        series = []
+        for p in packets:
+            dev_serial = obfuscate_serial(p['serial'])
+            for idx, c in enumerate(PACKET_FORMAT.channels(FILTER_DB_SCHEMA_COUNTERS)):
+                key = mklabel(p['serial'], c)
+                if key in self.map:
+                    tpl = self.map[key]
+                    dev_tag = tpl['tag']
```

Comment:
I've just pulled this down, trying to get it going with ECM1240 & trying to compare against the Brultech IA app the watt/Kwh are displayed fine in debug but the correct values don't go into Influxdb.  The main voltage & amps seem ok.
I also have gas pulses counting in Aux5 so would like to config the pulse unit is.
Another, nice feature from the Brultech Engine was the 'Virtual Tags' for Net Metering using Ch1/Ch2, need to brush up how this works ..... otta time tonight.